### ISSUE TITLE
Undefined property: stdClass::$resourceName

### ISSUE THREAD

**Issue by uicab2593** [EcommerceBBVA/BBVA-PHP#3]
Que tal compañeros, disculpen estoy haciendo una intergracion con VPOS y 3dSecure activado, pero al momento de hacer el

$charge = $bbva->charges->create($chargeRequest);

Me arroja un error en la biblioteca: 
Undefined property: stdClass::$resourceName
file: "../Bbva/data/BbvaApiResourceBase.php"
line: 161
message: "Undefined property: stdClass::$resourceName"
name: "PHP Notice"

Saben que podría ser? Estoy siguiendo el código ejemplo del README, espero puedan ayudarme, gracias!

**Comment by leozdim** [comment]
¿Tuviste problemas con un error similar al siguiente?
`Type: Error. Message: Call to undefined method BbvaApiResourceBase::getInstance()`

**Comment by BitQuarkApps** [comment]
Sí, recuerdo que modifiqué la api para poder usar meses sin intereses y devoluciones dentro de la tienda a la que le estoy dando mantenimiento

**Comment by sikios182** [comment]
Todavía no resuelven el bug, un año después sigue dando problemas con esa función.

**Comment by BitQuarkApps** [comment]
Creo que el código que puse lo arregla no?

**Comment by sikios182** [comment]
> Creo que el código que puse lo arregla no?

Si, pero solo lo he probado en sandbox, ¿te funcionó en productivo?

**Comment by BitQuarkApps** [comment]
Sí, el sitio donde se agregó esta SDK fue mvelectronica.com y actualmente se encuentra en producción, tanto pagos como reembolsos

**Comment by germaynbc** [comment]
Hola, tambien tengo el mismo problema, hice la recomendacion pero siempre que voy a procesar me dice Affiliation number 781500(este es el numero de afiliacion) is not valid , no se si es por el cambio de getResource

**Comment by BitQuarkApps** [comment]
> Hola, tambien tengo el mismo problema, hice la recomendacion pero siempre que voy a procesar me dice Affiliation number 781500(este es el numero de afiliacion) is not valid , no se si es por el cambio de getResource

No, el error es claro, tu número de afiliación es erróneo, revisa bien tus instancias de la clase para que veas que la clave es correcta, e incluso revisa que estés en modo sandbox porque al parecer ese número es sandbox

**Comment by germaynbc** [comment]
> > Hola, tambien tengo el mismo problema, hice la recomendacion pero siempre que voy a procesar me dice Affiliation number 781500(este es el numero de afiliacion) is not valid , no se si es por el cambio de getResource
> 
> No, el error es claro, tu número de afiliación es erróneo, revisa bien tus instancias de la clase para que veas que la clave es correcta, e incluso revisa que estés en modo sandbox porque al parecer ese número es sandbox

Si es correcto , verifique y todo muy gracias, una pregunta a usted tambien en pruebas le sale en el Sandbox , en los pagos sale el estatus Esperando Pago?

**Comment by germaynbc** [comment]
Hola buenas tardes, estoy con los casos de prueba, Cuando realizo el MSI, donde agrego "payments" => 6,
            "payments_type"=> "WITH_INTEREST", al array me da el siguiente error (Unrecognized field: payments_type) , me ayudaria con eso ? Muchas gracias

**Comment by sikios182** [comment]
> 
> 
> Hola buenas tardes, estoy con los casos de prueba, Cuando realizo el MSI, donde agrego "payments" => 6,
> "payments_type"=> "WITH_INTEREST", al array me da el siguiente error (Unrecognized field: payments_type) , me ayudaria con eso ? Muchas gracias

La documentación de los MSI define el objeto PaymentPlan de la siguiente manera:
{
   "payments": "6",
   "payments_type": "WITH_INTEREST",
   "deferred_months": "1"
}

En php sería:
'payment_plan' => array(
    'payments' => '6',
    'payments_type' => 'WITH_INTEREST',
    'deferred_months' => '1'
)

Donde "deferred_months" son los meses a los que el cliente empezará a pagar la compra.

¿Si estás mandando los 3 parámetros?

**Comment by germaynbc** [comment]
> payment_plan

Gracias, no estaba enviando dentro de payment_plan

**Comment by germaynbc** [comment]
En mi pasarela en prueba todo estana funcionando bien, ahora en prueba tambien cuando ingreso un tarjeta me muestra lo siguiente

![image](https://user-images.githubusercontent.com/30752935/96019864-26a22b00-0e1b-11eb-9100-cfd969d232e8.png)


¿Tiene alguna idea a que se debe? las credeciales las revise y si estan correctas

**Comment by sikios182** [comment]
> 
> 
> En mi pasarela en prueba todo estana funcionando bien, ahora en prueba tambien cuando ingreso un tarjeta me muestra lo siguiente
> 
> ![image](https://user-images.githubusercontent.com/30752935/96019864-26a22b00-0e1b-11eb-9100-cfd969d232e8.png)
> 
> ¿Tiene alguna idea a que se debe? las credeciales las revise y si estan correctas

Los códigos de error se encuentran dentro de la documentación que te provee BBVA dentro de su plataforma sandbox, la cual la encuentras en el siguiente enlace: 

https://docs.ecommercebbva.com/#c-digos-de-error

**Comment by germaynbc** [comment]
Si , se que es un usuario no autorizado. Pero tengo la duda si el usuario
que paga, o es el comercio. O que falla cuando nos marca ese error.

Muchas gracias, espero si me puede colaborar con el caso.

**Comment by sikios182** [comment]
Ya veo, creo que deberías contactarte con el equipo de plataformas especiales de BBVA, en lo personal no tuve ese error, pero me ayudaron con otras cosas y dudas que tuve. Igual te dejo el correo por si te interesa.

plataformas.especiales.mx@bbva.com

**Comment by germaynbc** [comment]
Quisiera algun soporte de porque :

3006 412 Precondition Failed La operación no esta permitida para este
cliente o esta transacción.


Si las credenciales estan correctas, entocnes a que otro factor me puede
dar ese error.

Muchas gracias

**Comment by Pabloaustria** [comment]
> No, el error es claro, tu número de afiliación es erróneo, revisa bien tus instancias de la clase para que veas que la clave es correcta, e incluso revisa que estés en modo sandbox porque al parecer ese número es sandbox



> Hola, tambien tengo el mismo problema, hice la recomendacion pero siempre que voy a procesar me dice Affiliation number 781500(este es el numero de afiliacion) is not valid , no se si es por el cambio de getResource
Me pasa los mismo, pudiste resolverlo?

**Comment by sikios182** [comment]
> > No, el error es claro, tu número de afiliación es erróneo, revisa bien tus instancias de la clase para que veas que la clave es correcta, e incluso revisa que estés en modo sandbox porque al parecer ese número es sandbox
> 
> 
> 
> > Hola, tambien tengo el mismo problema, hice la recomendacion pero siempre que voy a procesar me dice Affiliation number 781500(este es el numero de afiliacion) is not valid , no se si es por el cambio de getResource
> 
> 
> 
> 
> Me pasa los mismo, alguien sabe como resolverlo?

¿Nos puedes compartir el segmento de tu código donde instancias el pago?

**Comment by Pabloaustria** [comment]
> ¿Nos puedes compartir el segmento de tu código donde instancias el pago?
  ya pude resolverlo, a hora el detalle es que como se si el pago paso y fue rechazado y como extraigo los datos que devuelve

este es mi código de prueba

$bbva = Bbva::getInstance('mptdggroasfcmqs8plpy', 'sk_326c6d0443f6457aae29ffbd48f7d1be');
    Bbva::setSandboxMode(TRUE);
    $chargeRequest = array(
        'affiliation_bbva' => '781500',
        'amount' => 100,
        'description' => 'Cargo Venta',
        'currency' => 'MXN',
        'order_id' => 'FH-00003',
        'redirect_url' => 'https://sand-portal.ecommercebbva.com',
        'card' => array(
                'holder_name' => 'Juan Vazquez',
                'card_number' => '4242424242424242',
                'expiration_month' => '12',
                'expiration_year' => '28',
                'cvv2' => '842'),
        'customer' => array(
            'name' => 'Juan',
            'last_name' => 'Vazquez Juarez',
            'email' => 'juan.vazquez@empresa.com.mx',
            'phone_number' => '554-170-3567')
    );

    $charge = $bbva->charges->create($chargeRequest);
